ietf/meeting/views.py

```diff
@@ -2538,6 +2538,55 @@ def add_session_drafts(request, session_id, num):
                     'form': form,
                   })
 
+class SessionNotesAndRecordingsForm(forms.Form):
+    title = forms.CharField(max_length=255)
+    url = forms.URLField(label="Link to recording (YouTube only)")
+
+    def __init__(self, *args, **kwargs):
+        self.already_linked = kwargs.pop('already_linked')
+        super(self.__class__, self).__init__(*args, **kwargs)
+
+    def clean(self):
+        url = self.cleaned_data['url']
+        parsed_url = urlparse(url)
+        if parsed_url.hostname not in ['www.youtube.com', 'youtube.com', 'youtu.be', 'm.youtube.com', 'youtube-nocookie.com', 'www.youtube-nocookie.com']:
+            raise forms.ValidationError("Must be a YouTube URL")
+        problems = set(url).intersection(set(self.already_linked)) 
+        if problems:
+           raise forms.ValidationError("Already linked: %s" % ', '.join([d.name for d in problems]))
```

Comment:
This has a different semantic than the meetecho upload API, which can _update_ a link in case there was an error. We should think through what a chair would do if they entered the wrong link and wanted to fix it.